plugins/ginkgo/src/machine_backend.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 IFPEN-CEA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <alien/ginkgo/machine_backend.h>
+
+namespace Alien::Ginkgo
+{
+
+#ifdef ALIEN_PLUGIN_GINKGO_REF
+std::string ginkgo_executor::target_machine = "reference";
+#elif ALIEN_PLUGIN_GINKGO_OMP
+std::string ginkgo_executor::target_machine = "omp";
+#elif ALIEN_PLUGIN_GINKGO_CUDA
+std::string ginkgo_executor::target_machine = "cuda";
+#elif ALIEN_PLUGIN_GINKGO_HIP
+std::string ginkgo_executor::target_machine = "hip";
+#elif ALIEN_PLUGIN_GINKGO_DPCPP
+std::string ginkgo_executor::target_machine = "dpcpp";
+#else
+std::string ginkgo_executor::target_machine = "default";
+#endif
```

Comment:
Will disapear when implementing templated gingko wrapper.

Reply:
Yes.
This is also not well suited for manually handling data movement between two executors, if we want to add this feature.